cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/sink/DBSink.java

```diff
@@ -91,6 +92,13 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
                                                                                 getJDBCPluginId(),
                                                                                 PluginProperties.builder().build());
     Preconditions.checkArgument(jdbcDriverClass != null, "JDBC Driver class must be found.");
+    try {
+      Preconditions.checkArgument(tableExists(jdbcDriverClass), "Invalid table name %s", dbSinkConfig.tableName);
```

Comment:
Having a method name as `tableExists` with a non-table name as parameter is confusing. Can we instead rename this method `validateDBConnection`? We can move the precondition into the method then. `validateDBConnection` will then validate if you can connect to the DB and whether the table exists.